Documentation/development.md

```diff
@@ -0,0 +1,84 @@
+# CMO Development
+
+
+## Run CMO Locally
+### Assumptions
+* Should have a cluster running
+* Has setup Go development env
+
+### Build and Run CMO
+
+The `make` target `run-local` builds the `operator` and runs CMO locally
+
+```shell
+export KUBECONFIG=/path/to/kubeconfig/file
+make run-local
+```
+
+### Modify jsonnet
+
+**TL;DR:**
+1. build operator
+1. use `hack/local-cmo.sh` to run the operator
+1. modify assets/yamls
+1. backport the changes to jsonnet
+
+
+Sometimes we may not want to rebuild the operator. E.g. you may need
+to figure out how the stack behaves after applying a change to generated
+kubernetes manifest.  In other words, to modify the YAML in `assets/`
+directory directly without needing to change the `jsonnet/` or
+the `vendored` jsonnet files.
+
+In these cases it is easier to change those yaml files manually, run
+the operator (with those changed files) and later tweak `jsonnet`, than
+to figuring out which parts of jsonnet code you may need to change to
+fix an issue.
+
+It is easier to follow the following workflow
+#### Setup
+1. Start openshift cluster ( e.g. using `cluster-bot` )
+1. `export KUBECONFIG=/path/to/kubeconfig`
+1. Build operator - `make operator`
+
+#### Amend Assets
+1. Amend yaml files in `assets/` directory
+1. Run `hack/local-cmo.sh` (**does not** build the operator)
+1. ☝☝☝ _rinse and repeat_
+
+#### Finally:
+1. Port the changes back to `jsonnet` or vendored `jsonnet`
+
+
+
+## Updating vendored jsonnet code
```

Comment:
The usual way I do it which might be easier is:
```sh
cd jsonnet
jb update
# add only the dependencies that you want to update
git add -p -- jsonnetfile.lock.json
git checkout .
git reset HEAD -- jsonnetfile.lock.json
cd ..
make clean; make generate
git add -u
```

Reply:
The problem I think I find with `git add -p` is that  finding transitive dependencies of the **only** package that you want to update could be harder compared to the approach you suggested. 

I will add both and point this as the first approach to take.